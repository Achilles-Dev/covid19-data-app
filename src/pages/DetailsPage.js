import React from 'react';
import Regions from '../components/Regions';

const DetailsPage = () => (
  <div>
    <Regions />
  </div>
);

export default DetailsPage;
