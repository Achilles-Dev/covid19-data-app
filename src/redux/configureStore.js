import { combineReducers, legacy_createStore as createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import countriesReducer from './countries';

const rootReducer = combineReducers({
  countries: countriesReducer,
});

const composedEnhancer = composeWithDevTools(applyMiddleware(thunkMiddleware));

const store = createStore(rootReducer, composedEnhancer);

export default store;
