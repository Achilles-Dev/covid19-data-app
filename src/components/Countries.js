import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getAllCountries, getAllCountriesData } from '../redux/countries';
import Country from './Country';
import date from './Date';

const Countries = () => {
  const countries = useSelector((state) => state.countries.countries);
  const data = useSelector((state) => state.countries.data);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAllCountries());
  }, [dispatch]);

  useEffect(() => {
    dispatch(getAllCountriesData(date));
  }, [dispatch]);

  return (
    <div>
      <div className="main-header">
        <h2>WORLD STATS</h2>
        <p>{`Total confirmed: ${4}`}</p>
        <p>{`Total deaths: ${4}`}</p>
      </div>
      {
        countries.length > 0
          ? countries.map((country) => (
            <Country key={country.id} country={country} data={data} />
          ))
          : ''
      }
    </div>
  );
};

export default Countries;
