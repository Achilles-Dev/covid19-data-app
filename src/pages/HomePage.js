import React from 'react';
import Countries from '../components/Countries';

const HomePage = () => (
  <div className="main-page">
    <Countries />
  </div>
);

export default HomePage;
