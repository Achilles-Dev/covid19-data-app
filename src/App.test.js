import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import DetailsPage from './pages/DetailsPage';
import testStore from './redux/configureStore';
import HomePage from './pages/HomePage';
import Countries from './components/Countries';
import countriesReducer from './redux/countries';

const mockStore = configureStore([]);

describe('testing HomePage and Details page', () => {
  test('renders Homepage', () => {
    render(
      <Provider store={testStore}>
        <HomePage />
      </Provider>,
    );
    expect(screen.getByText('STATS BY COUNTRY')).toBeInTheDocument();
  });

  it('renders DetailsPage', () => {
    const details = render(
      <Provider store={testStore}>
        <DetailsPage />
      </Provider>,
    );
    expect(details).toMatchSnapshot();
  });
});

describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  const countries = [
    {
      id: 'us',
      name: 'US',
      links: [{
        href: '/api/countries/us/regions',
        rel: 'regions',
        type: 'GET',
      }],
    },
    {
      id: 'chad',
      name: 'Chad',
      links: [{
        href: '/api/countries/chad/regions',
        rel: 'regions',
        type: 'GET',
      }],
    },
  ];

  beforeEach(() => {
    store = mockStore({
      countries,
    });
  });

  it('should render with given state from Redux store', () => {
    store.dispatch = jest.fn();

    component = render(
      <Provider store={store}>
        <Countries />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
    expect(store.dispatch).toHaveBeenCalledTimes(1);
  });

  it('should return all countries', () => {
    const results = { countries: [...countries] };
    const getAllCountries = {
      type: 'covid19-data-app/countries/GET_ALL_COUNTRIES',
      payload: results,
    };

    expect(countriesReducer(undefined, getAllCountries)).toEqual({
      data: [],
      countries: [...countries],
    });
  });
});
