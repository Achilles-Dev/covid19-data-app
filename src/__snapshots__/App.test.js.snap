// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="main-header d-col-flex"
        >
          <h2>
            WORLD STATS
          </h2>
          <p>
            Confirmed cases: 0
          </p>
          <p>
            Deaths: 0
          </p>
        </div>
        <h3>
          STATS BY COUNTRY
        </h3>
        <div
          class="main-cards"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="main-header d-col-flex"
      >
        <h2>
          WORLD STATS
        </h2>
        <p>
          Confirmed cases: 0
        </p>
        <p>
          Deaths: 0
        </p>
      </div>
      <h3>
        STATS BY COUNTRY
      </h3>
      <div
        class="main-cards"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing HomePage and Details page renders DetailsPage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="region"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="region"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing HomePage and Details page renders Homepage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-page"
      >
        <div
          class="main"
        >
          <div
            class="main-header d-col-flex"
          >
            <h2>
              WORLD STATS
            </h2>
            <p>
              Confirmed cases: 0
            </p>
            <p>
              Deaths: 0
            </p>
          </div>
          <h3>
            STATS BY COUNTRY
          </h3>
          <div
            class="main-cards"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-page"
    >
      <div
        class="main"
      >
        <div
          class="main-header d-col-flex"
        >
          <h2>
            WORLD STATS
          </h2>
          <p>
            Confirmed cases: 0
          </p>
          <p>
            Deaths: 0
          </p>
        </div>
        <h3>
          STATS BY COUNTRY
        </h3>
        <div
          class="main-cards"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
